app/models/gamification_leaderboard.rb

```diff
@@ -16,7 +21,9 @@ def self.scores_for(leaderboard_id)
     users = users.where("gamification_scores.date BETWEEN ? AND ?", leaderboard.from_date, leaderboard.to_date) if leaderboard.from_date.present?
     # calculate scores up to to_date if just to_date is present
     users = users.where("gamification_scores.date <= ?", leaderboard.to_date) if leaderboard.to_date != Date.today && !leaderboard.from_date.present?
-    users = users.select("users.id, users.username, users.uploaded_avatar_id, #{sum_sql}").group("users.id").order(total_score: :desc).limit(100)
+    users = users.select("users.id, users.username, users.uploaded_avatar_id, #{sum_sql}").group("users.id").order(total_score: :desc)
+    users = users.offset(users.find_index(last_user) + 1) if last_user
```

Comment:
Instead of using offset here, can you try passing the last user score and using a `HAVING SUM(COALESCE(gamification_scores.score, 0)) < :last_user_score` ?

That will make this a lot faster.